docs/GettingStarted.md

```diff
@@ -119,180 +101,157 @@ Re-running generation now creates a file containing random strings that match th
+    "fields": [{ "name": "firstName", "type": "firstname" }],
+    "constraints": []
 }
 ```
 
-With the change in the `firstname` regex, the output is now starting to look quite realistic:
+Running the profile now gives a random list of first names.
 
 ```
-firstName,age
-"John",31
-"Mary",73
-"Jon",44
-"John",24
-"Jon",43
-"Jon",18
-"Jon",5
-"John",82
-"John",19
-"Jon",43
-[...]
+firstName
+Zac
+Sophie
+Millie
+Annabel
+Muhammad
+Ruaridh
 ```
 
-Finally, we'll add a field for National Insurance number. In this case, the constraints applied to the field ensure that it only has a value if the age is greater than or equal to 16:
+This is looking good but now we will want to add some more fields to get some more interesting data.
+
+First we'll expand the example profile to add a new `age` field, a not-null integer in the range 1-99:
 
 ```json
 {
     "fields": [
-        { "name": "firstName", "type": "string" },
-        { "name": "age", "type": "integer"  },
-        { "name": "nationalInsurance", "type": "string"  }
+      { "name": "firstName", "type": "firstname" },
+      { "name": "age", "type": "integer" }
     ],
     "constraints": [
-                { "not": { "field": "firstName", "is": "null" } },
-                {
-                    "field": "firstName",
-                    "matchingRegex": "(Joh?n|Mar[yk])"
-                },
-                { "field": "age", "greaterThan": 0 },
-                { "field": "age", "lessThan": 100 },
-                { "not": { "field": "age", "is": "null" } },
-                {
-                    "if": {
-                        "field": "age",
-                        "greaterThanOrEqualTo": 16
-                    },
-                    "then": {
-                        "allOf": [
-                            {
-                                "field": "nationalInsurance",
-                                "matchingRegex": "[A-CEGHJ-PR-TW-Z]{1}[A-CEGHJ-NPR-TW-Z]{1}[0-9]{6}[A-DFM]{0,1}"
-                            },
-                            {
-                                "not": {
-                                    "field": "nationalInsurance",
-                                    "is": "null"
-                                }
-                            }
-                        ]
-                    },
-                    "else": {
-                        "field": "nationalInsurance",
-                        "is": "null"
-                    }
-                }
+        { "field": "age", "greaterThan": 0 },
+        { "field": "age", "lessThan": 100 }
     ]
 }
 ```
 
-The above profile yields the following output:
+Next, we'll add some conditional logic to give some of our users a number plate. Lets add a `numberPlate` string to the list of fields.
 
+Fields are non-nullable by default, however, you can indicate that a field is nullable.  As we only want some users to have numbers plates, we should mark the `numberPlate` field as [`nullable`](UserGuide.md#nullable).
```

Comment:
Should be "we only want some users to have number plates", not "numbers plates"

Reply:
nice catch